internal/pkg/deploy/cloudformation/stack/workload.go

```diff
@@ -113,6 +113,20 @@ func (i ECRImage) GetLocation() string {
 	return fmt.Sprintf("%s:%s", i.RepoURL, "latest")
 }
 
+// GetSidecarLocation returns the sidecar container ECR image URI.
+// Prefer referring the image using git short commit id as <sidecarname>-<gitshortcommitid>.
+// Otherwise, each image after a push to ECR will get a digest and we refer to the image via the digest.
+// Finally, if no digest or tag is present, this occurs with the "package" commands, we default to the <sidecarname>-<latest> tag.
+func (i ECRImage) GetSidecarLocation(name string) string {
```

Comment:
This method is really similar to `GetLocation` above. 

I wonder if we could have removed the two different methods if `ECRImage` like this:
```go
type ECRImage struct {
	RepoURL  string // RepoURL is the ECR repository URL the container image should be pushed to.
	Tag string // Tag is the container image's unique tag.
	Digest   string // The image digest.
	ContainerName string // The container name. 
	MainContainerName string // The workload's container name.
}
```
The client would pass in both of the names to `ECRImage` in `workloadDeployer.runtimeConfig` and insider `GetLocation` we will check if the two names are the same or not and set things accordingly.

Reply:
Also if you don't mind renaming the methods to just `URL` 🥺